### PR TITLE
feat: add support for google_oauth2 token. Related to issue: https://…

### PR DESCRIPTION
**Pull request by anjali9791**
…github.com/goto/guardian/issues/168

### CODE REVIEW

File: pkg/http/http.go [hunk 1]
```diff
@@ -113,18 +132,35 @@ func (c *HTTPClient) setAuth(req *http.Request) {
...
-	req, err := http.NewRequest(http.MethodGet, c.config.URL, nil)
+	method := "GET"
+	if c.config.Method != "" {
+		method = c.config.Method
+	}
+	var body []byte
+	if c.config.Method == "POST" {
+		var err error
+		if c.config.Body == "" {
+			return nil, fmt.Errorf("body is required for POST method")
```
Comment (Gadigeppa-J) [inline pkg/http/http.go:143]: Payload is not mandatory for http post, this check is not required

File: pkg/http/http.go [hunk 2]
```diff
@@ -35,11 +37,12 @@ type HTTPAuthConfig struct {
...
 type HTTPClientConfig struct {
-	URL     string            `mapstructure:"url" json:"url" yaml:"url" validate:"required,url"`
-	Headers map[string]string `mapstructure:"headers,omitempty" json:"headers,omitempty" yaml:"headers,omitempty"`
-	Auth    *HTTPAuthConfig   `mapstructure:"auth,omitempty" json:"auth,omitempty" yaml:"auth,omitempty" validate:"omitempty,dive"`
-
-	HTTPClient *http.Client `mapstructure:"-" json:"-" yaml:"-"`
+	URL        string            `mapstructure:"url" json:"url" yaml:"url" validate:"required,url"`
+	Headers    map[string]string `mapstructure:"headers,omitempty" json:"headers,omitempty" yaml:"headers,omitempty"`
+	Auth       *HTTPAuthConfig   `mapstructure:"auth,omitempty" json:"auth,omitempty" yaml:"auth,omitempty" validate:"omitempty,dive"`
+	Method     string            `mapstructure:"method,omitmepty" json:"method,omitempty" yaml:"method,omitempty"`
```
Comment (Gadigeppa-J) [inline pkg/http/http.go:43]: typo 'omitmepty'

File: pkg/http/http.go [hunk 3]
```diff
@@ -113,18 +132,35 @@ func (c *HTTPClient) setAuth(req *http.Request) {
 }
 
 func (c *HTTPClient) MakeRequest(ctx context.Context) (*http.Response, error) {
```
Comment (Gadigeppa-J) [inline pkg/http/http.go:134]: Please add test case for covering POST method

File: pkg/http/http.go [hunk 4]
```diff
@@ -62,25 +65,33 @@ func NewHTTPClient(config *HTTPClientConfig) (*HTTPClient, error) {
...
-		ts, err := idtoken.NewTokenSource(ctx, config.Auth.Audience, idtoken.WithCredentialsJSON(creds))
-		if err != nil {
-			return nil, err
+		if config.Auth.Type == "google_idtoken" {
+			ts, err := idtoken.NewTokenSource(ctx, config.Auth.Audience, idtoken.WithCredentialsJSON(creds))
+			if err != nil {
+				return nil, err
+			}
+			httpClient = oauth2.NewClient(ctx, ts)
+		} else if config.Auth.Type == "google_oauth2" {
```
Comment (Gadigeppa-J) [inline pkg/http/http.go:88]: Please add test case for covering google_oauth2 auth type

**Review by Gadigeppa-J — CHANGES_REQUESTED** [review]

File: pkg/http/http.go (hunk 1 above)
Comment (anjali9791) [reply pkg/http/http.go:143]: done

File: pkg/http/http.go (hunk 3 above)
Comment (anjali9791) [reply pkg/http/http.go:133]: Done

File: pkg/http/http.go (hunk 4 above)
Comment (anjali9791) [reply pkg/http/http.go:93]: Done

**Review by Gadigeppa-J — APPROVED** [review]
lgtm

File: pkg/http/http.go [hunk 5]
```diff
@@ -62,25 +70,32 @@ func NewHTTPClient(config *HTTPClientConfig) (*HTTPClient, error) {
...
+			var err error
+			creds, err = decodeCredentials(config.Auth.CredentialsJSONBase64)
 			if err != nil {
-				return nil, fmt.Errorf("decoding credentials_json_base64: %w", err)
+				return nil, err
 			}
-			creds = v
 		default:
-			return nil, fmt.Errorf("missing credentials for google_idtoken auth")
+			return nil, fmt.Errorf("missing credentials for google_idtoken or  google_oauth2 auth")
```
Comment (rahmatrhd) [inline pkg/http/http.go:83]:
```suggestion
			return nil, fmt.Errorf("missing credentials for %q auth type", config.Auth.Type)
```

**Review by rahmatrhd — APPROVED** [review]

**Pull request merged**